packages/calypso-e2e/src/lib/blocks/file-block.ts

```diff
@@ -0,0 +1,66 @@
+import { Page, ElementHandle, Response, Frame } from 'playwright';
+
+const selectors = {
+	block: '.wp-block-file',
+	fileInput: '.components-form-file-upload input[type="file"]',
+	loadingAnimation: '.components-animate__loading.is-transient',
+};
+
+/**
+ * Represents the File block.
+ */
+export class FileBlock {
+	static blockName = 'File';
+	block: ElementHandle;
+
+	/**
+	 * Constructs an instance of this block.
+	 *
+	 * @param {ElementHandle} block Handle referencing the block as inserted on the Gutenberg editor.
+	 */
+	constructor( block: ElementHandle ) {
+		this.block = block;
+	}
+
+	/**
+	 * Uplaods the target file at the supplied path to WPCOM.
+	 *
+	 * @param {string} path Path to the file on disk.
+	 */
+	async upload( path: string ): Promise< void > {
+		// Obtain the page that this block resides on by first obtaining the frame then the page on which
+		// the frame belongs to.
+		const frame = ( await this.block.ownerFrame() ) as Frame;
+		const page = frame.page() as Page;
+
+		// The `input` element has a display:none style set and it cannot be found normally even if
+		// waitForSelector({state: 'hidden'}) is used.
+		const input = ( await this.block.$( selectors.fileInput ) ) as ElementHandle;
+		// Wait for the request complete instead of looking for the spinner and/or loading animation.
+		// Waiting on the animation class to be detached is not a reliable method for this block.
+		// It can lead to the filename placeholder text not being replaced with the uploaded file name.
+		await Promise.all( [
+			page.waitForResponse(
+				( response: Response ) => response.url().includes( 'media?' ) && response.status() === 200
+			),
+			input.setInputFiles( path ),
```

Comment:
This is similar to the convention used in [SupportComponent](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/lib/components/support-component.ts#L248-L252).